prov/usnic/src/usdf_fabric.c

```diff
@@ -101,6 +101,13 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 		return -FI_ENODATA;
 	}
 
+	/* it doesn't make sense to request both prefix and prefix2 */
+	if ((hints->mode & (FI_MSG_PREFIX | FI_MSG_PREFIX2)) ==
+			(FI_MSG_PREFIX | FI_MSG_PREFIX2)) {
+		USDF_INFO("Simultaneous use of FI_MSG_PREFIX and FI_MSG_PREFIX2 is invalid.\n");
+		return -EINVAL;
+	}
```

Comment:
This is one thing I didn't know how to handle. The usnic provider was reporting that it needed both `FI_MSG_PREFIX` and `FI_LOCAL_MR` if the hints structure was NULL. I changed it so that it now reports that it only needs `FI_LOCAL_MR` if the app gives a NULL hints structure. I could change it so that it now reports it needs `FI_LOCAL_MR` and `FI_MSG_PREFIX2`. 

@jsquyres What are your thoughts on this? 

@shefty you're right, it does make sense to accept both mode bits in the hints but I think it should prefer the newer mode bit. Does that sound reasonable to you?